zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -169,6 +219,15 @@ def test_920_pause_resume(self):
         """
         with self.pause_resume(self.services):
             logging.info("Testing pause resume")
+
+        logging.info("Wait till model is idle ...")
+        zaza.model.block_until_all_units_idle()
+
+        # If there are any blocekd mysql routers restart them.
+        self.restart_blocked_mysql_routers()
+        assert not self.get_blocked_mysql_routers(), (
+            "Should no longer be blocked mysql-router units")
```

Comment:
I'd be inclined to put this in the `restart_blocked_mysql_routers()` method, or perhaps have a another method called `ensure_no_blocked_mysql_routers()` that does the above three lines?